cmd/apps/vault_app.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) arkade author(s) 2022. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package apps
+
+import (
+	"fmt"
+
+	"github.com/alexellis/arkade/pkg"
+	"github.com/alexellis/arkade/pkg/apps"
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallVault() *cobra.Command {
+	var vault = &cobra.Command{
+		Use:          "vault",
+		Short:        "Install vault",
+		Long:         `Install vault`,
+		Example:      `  arkade install vault`,
+		SilenceUsage: true,
+	}
+
+	vault.Flags().StringP("namespace", "n", "vault", "The namespace to install the chart")
+	vault.Flags().StringArray("set", []string{}, "Use custom flags or override existing flags \n(example --set image.tag=1.11.2)")
+
+	vault.PreRunE = func(cmd *cobra.Command, args []string) error {
+		_, err := cmd.Flags().GetString("namespace")
+		if err != nil {
+			return err
+		}
+
+		_, err = cmd.Flags().GetStringArray("set")
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	vault.RunE = func(cmd *cobra.Command, args []string) error {
+		kubeConfigPath, _ := cmd.Flags().GetString("kubeconfig")
+		if err := config.SetKubeconfig(kubeConfigPath); err != nil {
+			return err
+		}
+
+		customFlags, _ := cmd.Flags().GetStringArray("set")
+		namespace, _ := cmd.Flags().GetString("namespace")
+
+		overrides := map[string]string{}
+		if err := config.MergeFlags(overrides, customFlags); err != nil {
+			return err
+		}
+
+		vaultOptions := types.DefaultInstallOptions().
+			WithNamespace(namespace).
+			WithHelmRepo("hashicorp/vault").
+			WithHelmURL("https://helm.releases.hashicorp.com").
+			WithOverrides(overrides).
+			WithKubeconfigPath(kubeConfigPath)
```

Comment:
I think its wise to add update repo here.
Example: https://github.com/alexellis/arkade/blob/master/cmd/apps/redis_app.go#L66

Reply:
added

Reply:
Please click "Resolve conversation" on these